src/dmd/mtype.d

```diff
@@ -3241,12 +3241,12 @@ extern (C++) final class TypeBasic : Type
         return size;
     }
 
-    override uint alignsize()
+    override uint alignsize() const
     {
-        return target.alignsize(this);
+        return target.alignsize(cast() this);
```

Comment:
You should update `Target::alignsize` as receiving a `const` parameter to avoid unnecessary cast.